### DISCUSSION TITLE
Can datafetcher be enabled by condition?

### DISCUSSION THREAD

**Discussion by jerry-ljh** [Netflix/dgs-framework#1346] [Q&A]
``` graphql
type Query {
    product(id: ID!): Product!
    productWithDataLoader(id: ID!): Product!
}

type Product {
    id: ID!
    name: String!
    shop: Shop!
}

type Shop {
    id: ID!
    name: String!
}
```
While testing shop field dataloader

```kotlin
@DgsData(parentType = DgsConstants.QUERY.Product, field = DgsConstants.PRODUCT.Shop)
fun shopLoader(dfe: DgsDataFetchingEnvironment): CompletableFuture<Shop> 
```
shoploader is always executed in product, productWithDataloader
But I want to apply it only to productWithDataLoader
I don't want to use shopLoader for product query, I want to call it up at once (ex join..)
Is it possible to apply shopLoader only to productWithDataLoader?

**Comment by srinivasankavitha** [comment]
You will likely have to apply that in your data fetcher implementation for
shop field, i.e. based on the root field (i.e. product or
productwithDataLoader) you will either invoke the data loader via "load" or
avoid do something else. Hope that helps.
